orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -423,7 +703,7 @@ Status TrainingRunner::EndTraining(IDataLoader* data_loader) {
     return Status::OK();
   }
 
-  if (data_loader) {
+  if (params_.num_pipeline_stages == 1 && data_loader) {
```

Comment:
>(params_.num_pipeline_stages == 1 &&  [](start = 5, length = 37)

Pipeline can use data_loader too, right?

Reply:
It's hard to judge which if a pipeline stage can `evaluate` a model. For example, if the evaluation depends on prediction, the evaluation can only happen in the last pipeline stage. I can change this to
```
  if (pipeline_stage_id == num_pipeline_stages - 1 && data_loader) {
```
What do you think?

Reply:
Still feel this evaluation part requires more thoughts. May we first finish the actual training part?